cachi2/core/package_managers/yarn/project.py

```diff
@@ -31,43 +40,48 @@ def cache_path(self) -> str:
 
         Fallback to the default path in case the configuration key is missing.
         """
-        return NotImplemented
+        return self._data.get("cacheFolder", "./.yarn/cache")
 
     @property
     def registry_server(self) -> str:
         """Get the globally configured registry server.
 
         Fallback to the default server in case the configuration key is missing.
         """
-        return NotImplemented
+        return self._data.get("npmRegistryServer", "https://registry.yarnpkg.com")
 
     @property
     def yarn_path(self) -> Optional[str]:
         """Path to the yarn script present in this directory."""
-        return NotImplemented
+        return self._data.get("yarnPath")
 
     @property
     def yarn_version(self) -> Optional[str]:
         """Yarn version used in this project.
 
         Extracted from the contents of the yarnPath configuration, if it is present.
         """
-        return NotImplemented
+        yarn_path = self.yarn_path
+        return re.search(r"yarn-(.*).cjs", yarn_path).group(1)
```

Comment:
Won't we hit an issue here if self.yarn_path is None?

Reply:
yup

Reply:
And also if the pattern doesn't match and re.search returns None (mypy caught both)

Reply:
TBH I feel like this logic doesn't really belong in YarnRC but in the code for setting yarn version

Reply:
ok, I'm gonna remove `yarn_version` property